src/util.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @import * as Ucanto from '@ucanto/interface'
+ * @import { Locator } from '@web3-storage/blob-fetcher'
+ */
+
+/**
+ * Returns a {@link Locator} which locates content only from a specific Space,
+ * by simply filtering the results of another {@link Locator}.
+ *
+ * @param {Locator} locator
+ * @param {Ucanto.DID[]} spaces
+ * @returns {Locator}
+ */
+export const spaceScopedLocator = (locator, spaces) => ({
+  locate: async (digest) => {
+    const locateResult = await locator.locate(digest)
+    if (locateResult.error) {
+      return locateResult
+    } else {
+      return {
+        ok: {
+          ...locateResult.ok,
+          site: locateResult.ok.site.filter(
+            (site) => site.space && spaces.includes(site.space)
```

Comment:
The space should be passed to the locator and used in the query to indexing service.

Reply:
Whoops, I forgot to switch this one to the new `locator.scopeToSpaces()`. Pushed!